internal/pkg/aws/session/session.go

```diff
@@ -30,11 +30,30 @@ func userAgentHandler() request.NamedHandler {
 	}
 }
 
-// Factory holds methods to create sessions.
-type Factory struct{}
+// Provider provides methods to create sessions.
+// Once a session is created, it's cached locally so that the same session is not re-created.
+type Provider struct {
+	defaultSess  *session.Session
+	regionCache  map[string]*session.Session
+	profileCache map[string]*session.Session
+	roleCache    map[string]*session.Session
+}
+
+// NewProvider initializes a new session Provider with empty caches.
+func NewProvider() *Provider {
+	return &Provider{
+		regionCache:  make(map[string]*session.Session),
+		profileCache: make(map[string]*session.Session),
+		roleCache:    make(map[string]*session.Session),
+	}
+}
 
 // Default returns a session configured against the "default" AWS profile.
-func (f *Factory) Default() (*session.Session, error) {
+func (p *Provider) Default() (*session.Session, error) {
```

Comment:
Should this function be thread-safe, just in case we might want to use goroutines in the future? Maybe use https://golang.org/pkg/sync/#Once?

Reply:
Same with all the maps we have in the `Provider` struct. Might want to use https://golang.org/pkg/sync/#RWMutex if we want to get fancy or to start simple, just use https://golang.org/pkg/sync/#Mutex

Reply:
Removed the caches after our discussion, I'll use the same idea and not use `sync.Once` until we have a need of running commands in parallel